pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -258,6 +258,8 @@ func (s *DiscoveryServer) updateClusterInc(push *model.PushContext, clusterName
 	if cnt == 0 {
 		push.Add(model.ProxyStatusClusterNoInstances, clusterName, nil, "")
 		//adsLog.Infof("EDS: no instances %s (host=%s ports=%v labels=%v)", clusterName, hostname, p, labels)
+	} else {
+		push.Delete(model.ProxyStatusClusterNoInstances, clusterName)
```

Comment:
See previous comment. this reads as if count==0, add somecluster, else delete some cluster.. which is not what the code is doing.